### ISSUE TITLE
Export List/Tuple type inputs with dynamic size

### ISSUE THREAD

**Issue by MaxeeCR** [pytorch/pytorch#112148]
### Background

I have a pytorch model with multiple inputs as below:
```
class MyModel():
  ...
  def forward(self, input):
  ""
  input: [object_batch_signals, object_batch_location, object_batch_sequences, agents_batch_signals,  agents_batch_location, agents_batch_category]
  object_batch_signals    (Tensor):         (B, C)
  object_batch_location   (Tensor):         (B, 2)
  object_batch_sequences  (Tensor):         (B, 3, 224, 224)
  agents_batch_signals    (List(Tensor)):   [B, Tensor(num_of_agents, C) or None]
  agents_batch_location   (List(Tensor)):   [B, Tensor(num_of_agents, 2) or None]
  agents_batch_category   (List(Tensor)):   [B, Tensor(num_of_agents, 1) or None]
  """
  ...
```
### About input
'object_batch_signals', 'object_batch_location', 'object_batch_sequences' are **normal `Tensor` type**
'agents_batch_signals',  'agents_batch_location', 'agents_batch_category' are `List` type which **include multiple `Tensor`s with difference shapes**.   

### About axis
* Dynamic axis
  * `B`: batch-size
  * `num_of_agents`: number of objects
* Static axis
  * `C`: channel size of feature 

### Question
Model has been trained successfully. But I don't know how to set inputs' shape when converting **pytorch->onnx->tensorrt**.
```
torch.onnx.export(
  model=model.cpu(),
  args=inputs,
  f=out_onnx_path,
  export_params=True,
  verbose=False,
  opset_version=opset_version,
  input_names=input_names,
  output_names=output_names,
  do_constant_folding=True,
  dynamic_axes=dynamic_axes)   <-----
```

Normal `dynamic_axes` for `Tensor` input should be like: 
```
dynamic_axes = {
  'object_batch_signals': {0: 'batch_size'},
  'object_batch_location': {0: 'batch_size'},
  'object_batch_sequences': {0: 'batch_size'},
  ...
  'output': {0: 'batch_size'}
}
```
But 'agents_batch_signals',  'agents_batch_location', 'agents_batch_category' are `List` type.

Could anyone take a look and help? Thanks!

## System Info
Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/main/torch/utils/collect_env.py)
```
Collecting environment information...
PyTorch version: 2.0.0
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.2 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: version 3.25.0
Libc version: glibc-2.31

Python version: 3.8.18 (default, Sep 11 2023, 13:40:15)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.4.0-81-generic-x86_64-with-glibc2.17
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 3090
GPU 1: NVIDIA GeForce RTX 3090

Nvidia driver version: 525.116.04
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.2.1
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   46 bits physical, 57 bits virtual
CPU(s):                          64
On-line CPU(s) list:             0-63
Thread(s) per core:              2
Core(s) per socket:              16
Socket(s):                       2
NUMA node(s):                    2
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           106
Model name:                      Intel(R) Xeon(R) Gold 6346 CPU @ 3.10GHz
Stepping:                        6
Frequency boost:                 enabled
CPU MHz:                         3600.000
CPU max MHz:                     3600.0000
CPU min MHz:                     800.0000
BogoMIPS:                        6200.00
Virtualization:                  VT-x
L1d cache:                       1.5 MiB
L1i cache:                       1 MiB
L2 cache:                        40 MiB
L3 cache:                        72 MiB
NUMA node0 CPU(s):               0-15,32-47
NUMA node1 CPU(s):               16-31,48-63
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Enhanced IBRS, IBPB conditional, RSB filling
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq rdpid md_clear pconfig flush_l1d arch_capabilities

Versions of relevant libraries:
[pip3] numpy==1.24.4
[pip3] onnx==1.14.1
[pip3] onnx-graphsurgeon==0.3.12
[pip3] onnxruntime-gpu==1.16.1
[pip3] torch==2.0.0
[pip3] torchaudio==2.0.0
[pip3] torchvision==0.15.0
[pip3] torchviz==0.0.2
[pip3] triton==2.0.0
[conda] blas                      1.0                         mkl  
[conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
[conda] mkl                       2023.1.0         h213fc3f_46343  
[conda] mkl-service               2.4.0            py38h5eee18b_1  
[conda] mkl_fft                   1.3.8            py38h5eee18b_0  
[conda] mkl_random                1.2.4            py38hdb19cb5_0  
[conda] numpy                     1.24.4                   pypi_0    pypi
[conda] pytorch                   2.0.0           py3.8_cuda11.7_cudnn8.5.0_0    pytorch
[conda] pytorch-cuda              11.7                 h778d358_5    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torchaudio                2.0.0                py38_cu117    pytorch
[conda] torchtriton               2.0.0                      py38    pytorch
[conda] torchvision               0.15.0               py38_cu117    pytorch
[conda] torchviz                  0.0.2                    pypi_0    pypi
```

**Comment by zhangvia** [comment]
i encountered a similar question. i got some inputs are none or dynamic shape tensors. did you solve it?

**Comment by thiagocrepaldi** [comment]
closing as there is no repro nor feedback from op

try the new onnx exporter api: torch.onnx.dynamo_export
